app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/room/ui/LiveDataRowPresenter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package android.support.v17.leanback.supportleanbackshowcase.app.room.ui;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.ViewModelProviders;
+import android.support.annotation.Nullable;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.SampleApplication;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.adapter.ListAdapter;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.db.entity.VideoEntity;
+import android.support.v17.leanback.supportleanbackshowcase.app.room.viewmodel.VideosInSameCategoryViewModel;
+import android.support.v17.leanback.widget.ListRow;
+import android.support.v17.leanback.widget.ListRowPresenter;
+import android.support.v17.leanback.widget.RowPresenter;
+import android.support.v4.app.FragmentActivity;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * The presenter for live data row
+ */
+public class LiveDataRowPresenter extends ListRowPresenter{
+
+    private ListRow mRow;
+    private List<StartEntranceListener> mEntranceListeners;
+    private VideosInSameCategoryViewModel mViewModel;
```

Comment:
each ViewHolder has it's own LiveData, Presenter is singleton.
Subclass ListRowPresenter.ViewHolder

Reply:
done